core/pkg/evaluator/json.go

```diff
@@ -301,19 +307,24 @@ func resolve[T constraints](ctx context.Context, reqID string, key string, conte
 func (je *Resolver) evaluateVariant(ctx context.Context, reqID string, flagKey string, evalCtx map[string]any) (
 	variant string, variants map[string]interface{}, reason string, metadata map[string]interface{}, err error,
 ) {
-	flag, metadata, ok := je.store.Get(ctx, flagKey)
-	if !ok {
+
+	var selector store.Selector
+	s := ctx.Value(store.SelectorContextKey{})
+	if s != nil {
+		selector = s.(store.Selector)
+	} else {
+		selector = store.NewSelector("")
+	}
+	if s != nil {
+		selector = s.(store.Selector)
+	}
```

Comment:
L318 - L320 seems duplicate.

Also, this can be written as 
```
selector := store.NewSelector("")
if s != nil {
   selector = s.(store.Selector)
}
```